homeassistant/components/starlink/coordinator.py

```diff
@@ -53,7 +56,10 @@ async def _async_update_data(self) -> StarlinkData:
                 status = await self.hass.async_add_executor_job(
                     status_data, self.channel_context
                 )
-                return StarlinkData(*status)
+                location = await self.hass.async_add_executor_job(
```

Comment:
I'm not sure but does this raise if location data is unavailable e.g. not turned on in the app?

In that case I think it should be it's own try-block and add to the data so we don't destroy `status` if this is not available.

Reply:
It doesn't appear to raise, but it will if Starlink is unreachable (status would fail then too)

It's probably a good idea to separate them anyways, just in case there's some API change in the future

Reply:
This separation doesn't make any sense as it would result in the same thing.
I was more thinking that if location raises we could just "ignore" it and continue e.g. only return status with an empty location or something like that.

But if it doesn't raise (I think you can test by turning it off on your own device?) then just put it together again as it was (to avoid duplicated code)

Reply:
Reverted